driver/src/models/standing_order.rs

```diff
@@ -19,20 +19,26 @@ impl StandingOrder {
     pub fn get_orders(&self) -> &Vec<super::Order> {
         &self.orders
     }
+    pub fn num_orders(&self) -> usize {
+        self.orders.len()
+    }
+    pub fn empty(account_id: u16) -> StandingOrder {
+        models::StandingOrder::new(account_id, 0, vec![])
+    }
+    pub fn initialize_array(&self) -> [StandingOrder; models::NUM_RESERVED_ACCOUNTS] {
```

Comment:
I liked the idea of having a macro that can be used for all kinds of types more than having `unsafe` specifically for StandingOrder, but I guess it's not a big issue. 

https://github.com/JoshMcguigan/arr_macro is a crate that has this feature claiming 100% safe rust code.

Reply:
Could you elaborate here on the meaning of `unsafe`?

Reply:
Never mind, I will read about it here --> [Unsafe Rust](https://doc.rust-lang.org/book/ch19-01-unsafe-rust.html)

Reply:
Well, generally, I would agree. But I am not so sure whether I am happy about pulling another package in there with an 0.* release. If we need it this initialization in two places, I would probably also use the package, but for just one place, I would stick with the unsafe code

I think having this small unsafe code is not much of an issue. Unsafe just means that the compiler can't run all the usual checks.

Reply:
I don't think the version number is a good argument as they are completely arbitrary and have very little meaning.

> I think having this small unsafe code is not much of an issue.

It can easily be an argument next time that there is already a place with unsafe code so it's fine to add another (cf. [The Broken Window Theory](https://www.freecodecamp.org/news/pragmatic-programmer-broken-windows-6916998eecbe/))

I don't insist on changing it but wanted to point this out.

Reply:
Cute. I enjoyed reading about the _Broken Window Theory_